docs/concepts/what-is-a-failure.md

```diff
@@ -0,0 +1,137 @@
+
+## Activity Failure
+
+An Activity Failure is delivered to the Workflow Execution when an Activity fails.
+It contains information about the failure and the Activity Execution, for example the Activity Type and Id.
+The reason for the failure will be in the `cause` field.
+For example, if an Activity Execution times out, the `cause` will be a [Timeout Failure](#timeout-failure).
+
+- TypeScript: [`ActivityFailure`](https://typescript.temporal.io/api/classes/common.ActivityFailure)
+- Java: [`ActivityFailure`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/failure/ActivityFailure.html)
+- Go: [`ActivityError`](https://pkg.go.dev/go.temporal.io/sdk/temporal#ActivityError)
+- Python: [`ActivityError`](https://python.temporal.io/temporalio.exceptions.ActivityError.html)
+- Proto: [`ActivityFailureInfo`](https://api-docs.temporal.io/#temporal.api.failure.v1.ActivityFailureInfo) and [`Failure`](https://api-docs.temporal.io/#temporal.api.failure.v1.Failure)
+
+## Child Workflow Failure
+
+A Child Workflow Failure is delivered to the Workflow Execution when a Child Workflow Execution fails.
+It contains information about the failure and the Child Workflow Execution, for example the Workflow Type and Id.
+The reason for the failure will be in the `cause` field.
+
+- TypeScript: [`ChildWorkflowFailure`](https://typescript.temporal.io/api/classes/common.ChildWorkflowFailure)
+- Java: [`ChildWorkflowFailure`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/failure/ChildWorkflowFailure.html)
+- Go: [`ChildWorkflowExecutionError`](https://pkg.go.dev/go.temporal.io/sdk/temporal#ChildWorkflowExecutionError)
+- Python: [`ChildWorkflowError`](https://python.temporal.io/temporalio.exceptions.ChildWorkflowError.html)
+- Proto: [`ChildWorkflowExecutionFailureInfo`](https://api-docs.temporal.io/#temporal.api.failure.v1.ChildWorkflowExecutionFailureInfo) and [`Failure`](https://api-docs.temporal.io/#temporal.api.failure.v1.Failure)
+
+## Timeout Failure
+
+Represents the timeout of an [Activity](/application-development/features#activity-timeouts) or [Workflow](/application-development/features#workflow-timeouts) Execution.
+
+When an Activity times out, the last Heartbeat details it emitted is attached.
+
+- TypeScript: [`TimeoutFailure`](https://typescript.temporal.io/api/classes/common.TimeoutFailure)
+- Java: [`TimeoutFailure`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/failure/TimeoutFailure.html)
+- Go: [`TimeoutError`](https://pkg.go.dev/go.temporal.io/sdk/temporal#TimeoutError)
+- Python: [`TimeoutError`](https://python.temporal.io/temporalio.exceptions.TimeoutError.html)
+- Proto: [`TimeoutFailureInfo`](https://api-docs.temporal.io/#temporal.api.failure.v1.TimeoutFailureInfo) and [`Failure`](https://api-docs.temporal.io/#temporal.api.failure.v1.Failure)
+
+## Terminated Failure
+
+Used as the `cause` when a Workflow has been Terminated.
```

Comment:
Used where?

Reply:
I'd mention you'd get this when waiting on a child workflow from another workflow or waiting for a workflow using a client.